src/mips_emulator/state.rs

```diff
@@ -966,6 +966,34 @@ impl InstrumentedState {
         self.pre_hash_root = page_hash_root;
         self.state.load_registers(); // add to rtrace
     }
+
+    pub fn get_split_segments(&mut self, proof: bool) -> Vec<Segment> {
```

Comment:
@evan95s  for split_segment, we can change the last parameter to io.Write as what we do for load_segment.

Reply:
It is a little different with read functions because it needs a plenty of io.Writers and the size is unknown when we use this function. So just give a parameter of io.Write may not working. Maybe a construct function which generate a io.Write?

Reply:
I see. But I see you only read one segment each time you call get_split_segments?  My advise is avoiding to implement the same logic in multiple functions. you can let the  split_segment to call get_split_segments to fetch all the segments, and then write them into files.

Reply:
Here may still need a litter discuss about split_segment function.
If the caller want to provide the writer, it need to know which segment it will generate, however the caller could not know at now because the pre_segment_id is private. A easy way to solve this problem is just make the pre_segment_id public, but I worry if it makes the caller must know every detail if it want to use the function? So I temporary use a write constructor for now, but I want to know which is better?